.github/workflows/_build-wheel-release-upload.yml

```diff
@@ -12,9 +12,40 @@ on:
       PYPI_TOKEN:
         description: 'PyPI token'
         required: true
+      PAT_TOKEN:
+        description: 'GitHub Personal Access Token'
+        required: true
 
 jobs:
+  tag-check:
+    runs-on: ubuntu-latest
+    outputs:
+      tag_valid: ${{ steps.tag-check.outputs.tag_valid }}
+    steps:
+      - name: Extract and validate tag for (pre)-release
+        id: tag-check
+        run: |
+          TAG_NAME=${GITHUB_REF#refs/tags/}
+
+          # Check if the tag matches the pattern, e.g. 3.1.32 or 0.1.3rc0
+          if [[ "$TAG_NAME" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]] || [[ "$TAG_NAME" =~ ^[0-9]+\.[0-9]+\.[0-9]+rc[0-9]+$ ]]; then
+            echo "tag_valid=true" >> $GITHUB_OUTPUT
+          fi
+
+  privilege-check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check user role
+        run: |
+          if [ "${{ github.actor }}" != "sbillinge" ]; then
```

Comment:
Here, only `sbillinge` is able to run this CI otherwise the CI will fail automatically.